// Copyright (c) 2023 Boston Dynamics, Inc.  All rights reserved.
//
// Downloading, reproducing, distributing or otherwise using the SDK Software
// is subject to the terms and conditions of the Boston Dynamics Software
// Development Kit License (20191101-BDSDK-SL).

syntax = "proto3";

package bosdyn.api;
option go_package = "bosdyn/api/data_acquisition_service";

option java_outer_classname = "DataAcquisitionServiceProto";

import "bosdyn/api/data_acquisition.proto";

// The DataAcquisitionService is the main data acquisition service run on robot, which receives
// incoming requests and sends queries to all directory-registered DataAcquisitionPluginServices.
service DataAcquisitionService {
    // Trigger a data acquisition to save data and metadata to the data buffer.
    // Sent by the tablet or a client to initiate a data acquisition and buffering process.
    rpc AcquireData(AcquireDataRequest) returns (AcquireDataResponse);

    // Query the status of a data acquisition.
    rpc GetStatus(GetStatusRequest) returns (GetStatusResponse);

    // Get information from a Data Acquisition service; lists acquisition capabilities.
    rpc GetServiceInfo(GetServiceInfoRequest) returns (GetServiceInfoResponse);

    // Cancel an in-progress data acquisition.
    rpc CancelAcquisition(CancelAcquisitionRequest) returns (CancelAcquisitionResponse);

    // Request live data available from DAQ plugins during teleoperation.
    // Please use the other RPCs for typical data acquisition.
    rpc GetLiveData(LiveDataRequest) returns (LiveDataResponse);
}
